browser/scripts/account-controller.js

```diff
@@ -24,25 +24,24 @@ AccountController.prototype._bindEvents = function(el)
 	{
 		evt.preventDefault();
 		bootbox.hideAll();
-		that.openLoginModal();
+		that.openLoginModal(dfd);
 	});
 
 	$('a.signup', el).on('click', function(evt)
 	{
 		evt.preventDefault();
 		bootbox.hideAll();
-		that.openSignupModal();
+		that.openSignupModal(dfd);
 	});
 }
 
-AccountController.prototype.openLoginModal = function()
-{
-
-	var self = this;
+AccountController.prototype.openLoginModal = function(dfd) {
+	var dfd = dfd || when.defer()
```

Comment:
1) although this works I'm not a big fan of naming the variable the same as an argument
2) The same dfd is bound to multiple events, is there a danger that multiple events can possibly resolve the event multiple times, potentially returning a resolved dfd before intended?